README.md

```diff
@@ -635,6 +635,29 @@ function hello($name, LoopInterface $loop)
 hello('Tester', $loop);
 ```
 
+If you want to create a timeout rather than limiting the number of executions,
+you can track timers in an array and pass it by reference. This is useful when
+waiting for another method or service to be ready.
+
+```php
+$timers = [];
+$cancelTimers = static function () use (&$timers) {
+  array_map('\React\EventLoop\Loop::cancelTimer', $timers);
+  $timers = [];
+};
+$timers[] = Loop::addPeriodicTimer(5, function () use ($cancelTimers) {
+    if (someExternalService()) {
+      $cancelTimers();
+      echo 'someExternalService is ready' . PHP_EOL;
+    }
+    echo 'Waiting for someExternalService' . PHP_EOL;
+});
+$timers[] = Loop::addTimer(60*45, function () use ($cancelTimers) {
+    $cancelTimers();
+    echo 'Timed out waiting for someExternalService' . PHP_EOL;
+});
+```
```

Comment:
Hey @danepowell, looks greeat :+1:

I have one last thing. If `someExternalService()` triggers and the timers get cancelled you print out "someExternalService is ready" which it's supposed to do but you also print one more "Waiting for someExternalService". Wrapping the second `echo` in an else would prevent this from happening.

Once we're at it I'd like to take the chance to make this example a little bit simpler for the user to read, I had something like this in mind:
```php
$timer = $loop->addPeriodicTimer(5, function () use ($loop, &$timer, &$timeoutTimer) {
    if (someExternalService()) {
        echo 'someExternalService is ready' . PHP_EOL;
        $loop->cancelTimer($timeoutTimer);
        $loop->cancelTimer($timer);
    } else {
        echo 'Waiting for someExternalService' . PHP_EOL;
    }
});

$timeoutTimer = $loop->addTimer(60*45, function () use ($loop, $timer) {
    $loop->cancelTimer($timer);
    echo 'Timed out waiting for someExternalService' . PHP_EOL;
});
```
Still has the same functionality. What do you think about this?

Reply:
Good catch on the if/else.

While I agree your example is less complex in some ways, it also doesn't scale when you have more than one or two timers since you're cancelling every timer in every other timer. The `$cancelTimer` callback is intuitive to me and much simpler in that case.

Also I thought using a `$loop` object was discouraged (instead of using the static `Loop::`)?